OAK-Marketplace-Emotion-Recognition/emotion_analyzer.py

```diff
@@ -0,0 +1,284 @@
+                                   self.report_statistics_shape), \
+                           np.clip(self.report_statistics_shape - result_bar_size - y_offset, 0,
+                                   self.report_statistics_shape)
+        
+        for emotion, emotion_bar in self.bars.items():
+            if emotion == top_emotion:
+                continue
+            percent = self.get_current_emotion_percent(emotion)
+            text = f'{emotion.upper()} - {percent}%'
+            cur_progress = emotion_bar.progress
+            cur_bar = EmotionBar(size=result_bar_size, emotion_size=result_emotion_size)
+            cur_bar.update_bar(progress=cur_progress)
+
+            result_statistic = add_class_names_and_percents(result_statistic,
+                                                            [start_x, int(result_statistic.shape[1] - y_offset / 2)],
+                                                            text)
+            alpha_mask = cur_bar.alpha_mask
+            overlay_image_alpha(result_statistic, cur_bar.bar, start_x, start_y, alpha_mask)
+            start_x = np.clip(start_x - result_bar_size - result_bar_padding, 0, result_statistic.shape[0])
+        if save:
+            cv2.imwrite(save_path, result_statistic)
+        return Image.fromarray(result_statistic[:, :, ::-1])
+
+
+class EmotionBar:
+    def __init__(
+        self, background_color: tuple = (28, 28, 30),
+        progress_color: tuple = (145, 82, 225),
+        size: int = 80,
+        thickness: int = 10,
+        start_angle: int = -220,
+        end_angle: int = 40,
+        emotion: str = 'happy',
+        update_count: int = 0,
+        progress: float = 0.0,
+        emotion_size: tuple = (25, 25),
+        emotion_path: str = 'images/emoji/happy.png',
+    ):
+        self.__update_count = update_count
+        self.__progress = progress
+        self.emotion_image = np.array(Image.open(emotion_path).resize(emotion_size))
+        self.size = size
+        self.emotion_size = emotion_size
+        self.emotion = emotion
+        self.background_color = background_color
+        self.progress_color = progress_color
+        self.thickness = thickness
+        self.start_angle = start_angle
+        self.end_angle = end_angle
+        self.bar = np.zeros((self.size, self.size, 3), dtype=np.uint8)
+        self.alpha_mask = np.ones((self.size, self.size, 3), dtype=np.uint8)
+        self.width, self.height, self.radius, self.center, self.axes = self.calculate_bar_parameters()
+
+    @property
+    def update_count(self):
+        return self.__update_count
+
+    @property
+    def progress(self):
+        return self.__progress
+
+    @update_count.setter
+    def update_count(self, value):
+        self.__update_count = value
+
+    @progress.setter
+    def progress(self, value):
+        self.__progress = value
+
+    def calculate_bar_parameters(self):
+        height, width = self.bar.shape[0:2]
```

Comment:
Looks like `height, width` is `self.size` because of `self.bar = np.zeros((self.size, self.size, 3), dtype=np.uint8)`